packages/timeline-state-resolver/src/integrations/vmix/index.ts

```diff
@@ -1454,3 +1432,131 @@ export interface VMixAudioChannel {
 	meterF2: number
 	headphonesVolume: number
 }
+
+/**
+ * Handles pre-loading of inputs.
+ */
+class VMixInputHandler extends EventEmitter {
+	// how long to let an input linger before removing it
+	private TTL = 1000
+
+	/** Tracks the actual state in vmix of which inputs are loaded */
+	private _inputs: Map<
+		string,
+		{
+			type: string
+			name: string
+			/** Is set to true when the commands to add the input has been set. */
+			addedCommandSent: boolean
+			/** Is set to something when the input is scheduled to be removed. */
+			removeTime?: number
+		}
+	> = new Map()
+
+	private _updateTimeout: NodeJS.Timeout | undefined = undefined
+
+	constructor(private _vmix: VMixDevice) {
```

Comment:
Instead of passing the entire class to a subclass you should just pass callbacks for adding commands and getting the time. That will also make it easier to add unit tests in the future and refactor the device integration.

Reply:
Yes, but I still want to be careful with adding methods on these device classes as they are exposed to anyone consuming the TSR.